packages/cookie-consent/src/CookieConsentProvider/CookieConsentProvider.tsx

```diff
@@ -0,0 +1,245 @@
+import cookie from 'cookie'
+import type { ReactNode } from 'react'
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react'
+import { uniq } from '../helpers/array'
+import { stringToHash } from '../helpers/misc'
+import type { CategoryKind, Config, Consent, Integrations } from './types'
+import { useSegmentIntegrations } from './useSegmentIntegrations'
+
+const COOKIE_PREFIX = '_scw_rgpd'
+const HASH_COOKIE = `${COOKIE_PREFIX}_hash`
+
+// Appx 13 Months
+const CONSENT_MAX_AGE = 13 * 30 * 24 * 60 * 60
+// Appx 6 Months
+const CONSENT_ADVERTISING_MAX_AGE = 6 * 30 * 24 * 60 * 60
+
+const COOKIES_OPTIONS = {
+  sameSite: 'strict',
+  secure: true,
+  path: '/',
+} as const
+
+type Context = {
+  integrations: Integrations
+  needConsent: boolean
+  isSegmentAllowed: boolean
+  segmentIntegrations: { All: boolean } & Record<string, boolean>
+  categoriesConsent: Partial<Consent>
+  saveConsent: (categoriesConsent: Partial<Consent>) => void
+}
+
+const CookieConsentContext = createContext<Context | undefined>(undefined)
+
+export const useCookieConsent = (): Context => {
+  const context = useContext(CookieConsentContext)
+  if (context === undefined) {
+    throw new Error(
+      'useCookieConsent must be used within a CookieConsentProvider',
+    )
+  }
+
+  return context
+}
+
+export const CookieConsentProvider = ({
+  children,
+  isConsentRequired,
+  essentialIntegrations,
+  config,
+  cookiePrefix = COOKIE_PREFIX,
+  consentMaxAge = CONSENT_MAX_AGE,
+  consentAdvertisingMaxAge = CONSENT_ADVERTISING_MAX_AGE,
+  cookiesOptions = COOKIES_OPTIONS,
+}: {
+  children: ReactNode
+  isConsentRequired: boolean
+  essentialIntegrations: string[]
+  config: Config
+  cookiePrefix: string
+  consentMaxAge: number
+  consentAdvertisingMaxAge: number
+  cookiesOptions: {
+    sameSite: boolean | 'strict' | 'lax' | 'none' | undefined
+    secure: boolean
+    path: string
+  }
```

Comment:
Could we derive this type from `COOKIES_OPTIONS`, or create a separate type to be used in both places?